### PR TITLE
[TS] LPS-145396 - TESTS

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
Hi @tinatian

**TESTS of https://issues.liferay.com/browse/LPS-145396 - When we delete some types of objects in Liferay, we are not deleting their ResourcePermission records, leaving some orphan data in the ResourcePermission table.**

New pull request following your guidelines of your comment here https://github.com/liferay-core-infra/liferay-portal/pull/596#issuecomment-1098507497

Inside the `_smartDelete` method, (see https://github.com/liferay/liferay-portal/commit/92018c325ff8b42f8c2becbb30970bc8b2dca88f#diff-d987badaa7c5a1822c10382d9cfdc8711f3edad59b1b198cca0ffbd9790a7837R650 ) I had added a call to the `ResourcePermissionTestUtil.deleteResourcePermissions(persistedModel);` method because the **smartDelete** code calls the persistedModelLocalService code, so it is not deleting the related objects.

Inside the `ResourcePermissionTestUtil.deleteResourcePermissions(persistedModel);` method I am deleting all the ResourcePermission objects related to the deleted object.

After this, the **_orphanResourcePermissionDetection** https://github.com/liferay/liferay-portal/commit/92018c325ff8b42f8c2becbb30970bc8b2dca88f#diff-d987badaa7c5a1822c10382d9cfdc8711f3edad59b1b198cca0ffbd9790a7837R122 will detect all the orphan ResourcePermission that weren't deleted during the test execution.

**Test run:** in this PR https://github.com/jorgediaz-lr/liferay-portal/pull/477 I have added a new commit https://github.com/jorgediaz-lr/liferay-portal/pull/477/commits/70a604836f8173b44ad111e55acf192542866f40 that removes the `resourceLocalService.deleteResource` execution after JournalArticle and DLFileEntry deletion and my code successfully detects the inconsistence, see: https://github.com/jorgediaz-lr/liferay-portal/pull/477#issuecomment-1128983824

Let me know if you need more information about this PR

Regards,
Jorge

### CODE REVIEW

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

**Comment by jorgediaz-lr** [comment]
Hi @tinatian,

Do you have any update about this pull request?

Thank you

File: portal-test/src/com/liferay/portal/kernel/test/rule/DataGuardTestRuleUtil.java [hunk 1]
```diff
@@ -489,6 +509,111 @@ private static Closeable _installTransactionExecutor(
...
+
+			return false;
+		}
+		catch (NoSuchModelException noSuchModelException) {
+		}
+
+		return true;
+	}
+
+	private static void _orphanResourcePermissionDetection(
```
Comment (dantewang) [inline portal-test/src/com/liferay/portal/kernel/test/rule/DataGuardTestRuleUtil.java:567]: @tinatian @jorgediaz-lr I wonder if we really need a new method to do the detection.

The `_autoDeleteAndAssert` method calls `_autoDeleteLeftOvers` first (if `autoDelete == true`), and then re-captures the snapshot and does the assertion.

So, if we just skip `ResourcePermission` in `_autoDeleteLeftOvers`, but don't skip it in the assertion in `_autoDeleteAndAssert`, the left over `ResourcePermission` can still be detected.

Please see https://github.com/dantewang/liferay-portal/pull/2227/commits/f5d8890f7c021b280311621c052b31e61c1fab1d for what I mean.

I think this way does the same thing that Tina suggests in https://github.com/liferay-core-infra/liferay-portal/pull/596#issuecomment-1098507497

Comment (dantewang) [reply portal-test/src/com/liferay/portal/kernel/test/rule/DataGuardTestRuleUtil.java:567]: @jorgediaz-lr Could you check if there's any edge case that the above approach can't cover?

File: modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/closure/test/CTClosureFactoryImplTest.java [hunk 2]
```diff
@@ -194,8 +194,7 @@ public void tearDown() throws Exception {
 					"CTCollectionLocalServiceImpl",
 				LoggerTestUtil.WARN)) {
 
-			_ctCollectionLocalService.deleteCTCollection(
-				_ctCollection.getCtCollectionId());
+			_ctCollectionLocalService.deleteCTCollection(_ctCollection);
```
Comment (dantewang) [inline modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/closure/test/CTClosureFactoryImplTest.java:197]: @jorgediaz-lr Based on the sub tasks in LPS-145936, I think it's better to send this commit in a separate sub task, so it can be merged quickly.

**Review by dantewang — CHANGES_REQUESTED** [review]
@jorgediaz-lr Please see my comments and resend, thanks!

**Comment by dantewang** [comment]
ci:close

**Comment by jorgediaz-lr** [comment]
Hi @dantewang thanks for reviewing this PR, I will have a look to your comments and I will resend.

**Comment by jorgediaz-lr** [comment]
Sent a new PR here https://github.com/liferay-core-infra/liferay-portal/pull/1020
